core/modules/replica/ConfigurationTypes.cc

```diff
@@ -0,0 +1,79 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/ConfigurationTypes.h"
+
+using json = nlohmann::json;
+
+namespace {
+
+using namespace lsst::qserv::replica;
+    
+template <typename T>
+json paramToJson(T const& struct_,
+                 Configuration::Ptr const& config) {
+    return json({
+        {"parameter",   struct_.key},
+        {"value",       struct_.get(config)},
+        {"updatable",   struct_.updatable ? 1 : 0},
+        {"description", struct_.description}});
+}
+}
+
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+json ConfigurationGeneralParams::toJson(Configuration::Ptr const& config) const {
+
+    json result;
+
+    result.push_back(::paramToJson(requestBufferSizeBytes,      config));
+    result.push_back(::paramToJson(retryTimeoutSec,             config));
+    result.push_back(::paramToJson(controllerThreads,           config));
+    result.push_back(::paramToJson(controllerHttpPort,          config));
+    result.push_back(::paramToJson(controllerHttpThreads,       config));
+    result.push_back(::paramToJson(controllerRequestTimeoutSec, config));
+    result.push_back(::paramToJson(jobTimeoutSec,               config));
+    result.push_back(::paramToJson(jobHeartbeatTimeoutSec,      config));
+    result.push_back(::paramToJson(xrootdAutoNotify,            config));
+    result.push_back(::paramToJson(xrootdHost,                  config));
+    result.push_back(::paramToJson(xrootdPort,                  config));
+    result.push_back(::paramToJson(xrootdTimeoutSec,            config));
+    result.push_back(::paramToJson(databaseTechnology,          config));
+    result.push_back(::paramToJson(databaseHost,                config));
+    result.push_back(::paramToJson(databasePort,                config));
+    result.push_back(::paramToJson(databaseUser,                config));
+    result.push_back(::paramToJson(databasePassword,            config));
+    result.push_back(::paramToJson(databaseName,                config));
+    result.push_back(::paramToJson(databaseServicesPoolSize,    config));
+    result.push_back(::paramToJson(workerTechnology,            config));
+    result.push_back(::paramToJson(workerNumProcessingThreads,  config));
+    result.push_back(::paramToJson(fsNumProcessingThreads,      config));
+    result.push_back(::paramToJson(workerFsBufferSizeBytes,     config));
```

Comment:
does result offer any kind of `reserve` functionality like a `vector` (since it seems to borrow from the `vector` api...). And or static ctor `{...}` could be very cool.

Reply:
AFAIK, the `son` objects don't have method `reserve`. At least, I don't see this in the documentation:
https://github.com/nlohmann/json